packages/core-p2p/lib/peer.js

```diff
@@ -65,16 +66,29 @@ module.exports = class Peer {
    * @return {(Object|undefined)}
    */
   async postTransactions(transactions) {
-    return this.__post(
-      '/peer/transactions',
-      {
-        transactions,
-      },
-      {
-        headers: this.headers,
-        timeout: 8000,
-      },
-    )
+    const broadcast = async items =>
+      this.__post(
+        '/peer/transactions',
+        {
+          transactions: items,
+        },
+        {
+          headers: this.headers,
+          timeout: 8000,
+        },
+      )
+
+    try {
+      await broadcast(transactions)
+    } catch (err) {
+      if (err.response && err.response.status === 413) {
+        const items = chunk(transactions, err.response.data.error.allowed)
+
+        await Promise.all(items.map(item => broadcast(item)))
```

Comment:
Hm, are you sure this is a good idea? What if sender has 200 txs and receiver allows max 10, then that's 20 requests one after another. How about a little delay between the requests, like 200ms or something?

Reply:
I think that sender should be sending as much as possible as faster as possible. If that does not suit the receiver he should respond accordingly. Arbitrary delay on the sender side has the problems that 1. most of the time it will cause unnecessary slowdown and 2. maybe it will not meet the requirements on the receiver - thus be useless.

Reply:
@vasild 	what if receiver only allows 1 tx per request? that will hold up the sender node because it won't continue until it's finished, which could also affect other areas of the sending node.

Reply:
If that would be a problem, we can check if `err.response.data.error.allowed` is too low. But I do not think this is the case - the overhead for sending 1 transaction per request and 100 transactions per request is comparable (I did not do the exact calculations).

Reply:
This opens possible attack window, and also await will delay it.
This is critical p2p and nodes should be in sync with the same protocol. Either it recipient accepts the payload or not, they dont want to have a chat, and also not every pool is required to be the same.

Reply:
You're probably right, but I just don't like such loops and I'm also worried because the main problem with the network was the broadcasting, because of too many connections peers were blocking each other and the network loses quorum. I hope this will not cause the same issues.